shared/src/business/entities/cases/Case.ts

```diff
@@ -24,2501 +23,2441 @@ const {
-  obj.petitionPaymentDate = rawCase.petitionPaymentDate;
-  obj.petitionPaymentMethod = rawCase.petitionPaymentMethod;
-  obj.petitionPaymentStatus =
-    rawCase.petitionPaymentStatus || PAYMENT_STATUS.UNPAID;
-  obj.petitionPaymentWaivedDate = rawCase.petitionPaymentWaivedDate;
-  obj.preferredTrialCity = rawCase.preferredTrialCity;
-  obj.procedureType = rawCase.procedureType;
-  obj.receivedAt = rawCase.receivedAt || createISODateString();
-  obj.sealedDate = rawCase.sealedDate;
-  obj.status = rawCase.status || CASE_STATUS_TYPES.new;
-  obj.sortableDocketNumber =
-    rawCase.sortableDocketNumber ||
-    Case.getSortableDocketNumber(rawCase.docketNumber);
-  obj.trialDate = rawCase.trialDate;
-  obj.trialLocation = rawCase.trialLocation;
-  obj.trialSessionId = rawCase.trialSessionId;
-  obj.trialTime = rawCase.trialTime;
-  obj.useSameAsPrimary = rawCase.useSameAsPrimary;
-
-  obj.initialDocketNumberSuffix =
-    rawCase.initialDocketNumberSuffix || obj.docketNumberSuffix || '_';
-
-  if (rawCase.caseCaption) {
-    obj.initialCaption = rawCase.initialCaption || obj.caseCaption;
-  }
-
-  obj.hasPendingItems = obj.docketEntries.some(docketEntry =>
-    DocketEntry.isPending(docketEntry),
-  );
+  getErrorToMessageMap() {
+    return Case.VALIDATION_ERROR_MESSAGES;
+  }
+
+  assignFieldsForInternalUsers({ applicationContext, rawCase }) {
+    this.associatedJudge = rawCase.associatedJudge || CHIEF_JUDGE;
+    this.automaticBlocked = rawCase.automaticBlocked;
+    this.automaticBlockedDate = rawCase.automaticBlockedDate;
+    this.automaticBlockedReason = rawCase.automaticBlockedReason;
+    this.blocked = rawCase.blocked;
+    this.blockedDate = rawCase.blockedDate;
+    this.blockedReason = rawCase.blockedReason;
+    this.caseStatusHistory = rawCase.caseStatusHistory || [];
+    this.caseNote = rawCase.caseNote;
+    this.damages = rawCase.damages;
+    this.highPriority = rawCase.highPriority;
+    this.highPriorityReason = rawCase.highPriorityReason;
+    this.judgeUserId = rawCase.judgeUserId;
+    this.litigationCosts = rawCase.litigationCosts;
+    this.qcCompleteForTrial = rawCase.qcCompleteForTrial || {};
+
+    this.noticeOfAttachments = rawCase.noticeOfAttachments || false;
+    this.orderDesignatingPlaceOfTrial =
+      rawCase.orderDesignatingPlaceOfTrial || false;
+    this.orderForAmendedPetition = rawCase.orderForAmendedPetition || false;
+    this.orderForAmendedPetitionAndFilingFee =
+      rawCase.orderForAmendedPetitionAndFilingFee || false;
+    this.orderForFilingFee = rawCase.orderForFilingFee || false;
+    this.orderForCds = rawCase.orderForCds || false;
+    this.orderForRatification = rawCase.orderForRatification || false;
+    this.orderToShowCause = rawCase.orderToShowCause || false;
+
+    this.assignArchivedDocketEntries({ applicationContext, rawCase });
+    this.assignStatistics({ applicationContext, rawCase });
+    this.assignCorrespondences({ rawCase });
+  }
+
+  static VALIDATION_RULES = {
```

Comment:
Hey @codyseibert are the case validation rules better housed in the entityvalidationConstants file?

Reply:
I think there are some Validation rules which are shared between entities. Such as startDate, endDate validation rules could be reused by many entities and is common so gets extracted to entityValidationConstants. However if the validation rules are not shared anywhere then it should probably just live in the entity itself